include/exec/any_sender_of.hpp

```diff
@@ -0,0 +1,805 @@
+              }}};
+              return &__vtable_;
+            }
+        };
+        
+        class __env_t {
+          public:
+          __env_t(const __vtable* __vtable, void* __sender) noexcept
+          : __vtable_{__vtable}, __sender_{__sender} {
+          }
+          private:
+          const __vtable* __vtable_;
+          void* __sender_;
+
+          template <class _Tag, class... _As>
+              requires __callable<const __query_vtable<_SenderQueries>&, _Tag, void*, _As...>
+            friend auto tag_invoke(_Tag, const __env_t& __self, _As&&... __as)
+              noexcept(__nothrow_callable<const __query_vtable<_SenderQueries>&, _Tag, void*, _As...>)
+              -> __call_result_t<const __query_vtable<_SenderQueries>&, _Tag, void*, _As...> {
+              return __self.__vtable_->__queries()(_Tag{}, __self.__sender_, (_As&&) __as...);
+            }
+        };
+
+        class __t {
+         public:
+          using __id = __sender;
+          using completion_signatures = _Sigs;
+          using is_sender = void;
+
+          __t(const __t&) = delete;
+          __t& operator=(const __t&) = delete;
+
+          __t(__t&&) = default;
+          __t& operator=(__t&&) = default;
+
+          template <class _Sender>
+              requires (!__decays_to<_Sender, __t> &&
+                        sender_to<_Sender, __receiver_ref<_Sigs, _ReceiverQueries>>)
+            __t(_Sender&& __sndr)
+              : __storage_{(_Sender&&) __sndr} {}
+
+          __unique_operation_storage __connect(__receiver_ref_t __receiver) {
+            return __get_vtable(__storage_)->__connect_(__get_object_pointer(__storage_), 
+                                                        (__receiver_ref_t &&) __receiver);
+          }
+
+         private:
+          __unique_storage_t<__vtable> __storage_;
+
+          template <receiver_of<_Sigs> _Rcvr>
+            friend stdexec::__t<__operation<__t, std::decay_t<_Rcvr>, _ReceiverQueries>>
+            tag_invoke(connect_t, __t&& __self, _Rcvr&& __rcvr) {
+              return {(__t&&) __self, (_Rcvr&&) __rcvr};
+            }
+
+          friend __env_t tag_invoke(get_env_t, const __t& __self) noexcept {
+            return {__get_vtable(__self.__storage_), __get_object_pointer(__self.__storage_)};
+          }
+        };
+      };
+
+    template <class _R, class... _Args>
+    _R __return_value_(_R(*)(_Args...));
+
+    template <class _Sig>
+    using __return_value_t = decltype(__return_value_((_Sig*)nullptr));
+
+    template <class _Sigs = completion_signatures<set_value_t()>,
+              class _SchedulerQueries = __types<>,
+              class _SenderQueries = __types<>,
+              class _ReceiverQueries = __types<>>
+      class __scheduler {
+      public:
+        template <class _Scheduler>
+            requires (!__decays_to<_Scheduler, __scheduler>) // && scheduler<_Scheduler>)
```

Comment:
Seems to be a bad interaction with Catch2. This recursion is bothersome. I haven't  yet managed to track it down, but naturally it will need to be fixed.

Reply:
I think I fixed this in d4bdc56.